accepted/00000-ansible-integration.md

```diff
@@ -0,0 +1,304 @@
+
+This is an example of Ansible playbook:
+
+```yaml
+---
+- name: update web servers
+  hosts: webservers
+  remote_user: root
+
+  tasks:
+  - name: ensure apache is at the latest version
+    yum:
+      name: httpd
+      state: latest
+  - name: write the apache config file
+    template:
+      src: /srv/httpd.j2
+      dest: /etc/httpd.conf
+
+- name: update db servers
+  hosts: databases
+  remote_user: root
+
+  tasks:
+  - name: ensure postgresql is at the latest version
+    yum:
+      name: postgresql
+      state: latest
+  - name: ensure that postgresql is started
+    service:
+      name: postgresql
+      state: started
+```
+
+As you can see, this playbook is defining different tasks to execute on different "hosts" groups. Those groups are of course defined in the Ansible inventory (default at /etc/ansible/hosts).
+
+The execution of a given playbooks is done in the corresponding Ansible controller, which contains the inventory and the hosts definition.
+
+The playbooks available to apply on each Ansible controller, are the ones exposed by "ansible-gatherer" and ultimately, in the future, playbooks created and maintained in Uyuni (via future Ansible playbook catalog) that would be then pushed to the controller node.
+
+At the time of running a playbook from the Ansible controller, Uyuni is able to do it by reaching the controller and operating it using "salt-ssh" (in principle, the Ansible controller is not necessary a registered running minion). There are different approaches here:
+
+Option 1)
+- SaltSSH to Ansible controller located in Ansible controller) via Salt state (or execution module):
+
+```yaml
+execute_ansible_playbook:
+  ansible.playbooks:
+    - name: /some/path/in/my/ansible/controller/playbook_1.yaml
+```
+
+Option 2)
+- SaltSSH to Ansible controller to apply Salt state (playbook coming from Uyuni):
+
+```yaml
+execute_ansible_playbook:
+  ansible.playbooks:
+    - name: salt://ansible_playbooks/org_1/inventory-label/playbook-1.yaml
+```
+
+The option 1 is refers to executing a playbook which is already stored and maintained in the Ansible controller. On the other hand, option 2 would work in an scenario where the maintenance of the Playbooks are done in Uyuni, so we need to make the Ansible controller to get the generated playbook by Uyuni from the Salt file roots.
+
+The SSH credentials to use in both cases to reach the Ansible controller would be the same that "ansible-gatherer" is using.
+
+In case we allow "ansible-gatherer" to add an Ansible controller from a registered system, then playbooks execution can be triggered to the controller either using "salt-ssh" or via normal Salt minion job, depending on how the system is registered in Uyuni.
+
+
+### Execute Salt commands & states to Ansible systems
+The "ansiblegate" module of Salt also allows "salt-ssh" to reuse an Ansible inventory to reach those systems that are being managed by Ansible and execute Salt commands on it.
```

Comment:
Is ansiblegate limited to a plain-file inventory or can it consume any inventory from a plugin?

https://docs.ansible.com/ansible/latest/user_guide/intro_dynamic_inventory.html

Reply:
You can pass the inventory you want to use, so you have control of the `-i` option of Ansible. IIUC it should work with dynamic inventories if you pass the module as argument of `ansible -i`